### PR TITLE
feat(RHINENG-7070) GET /hosts DB implementation (Part 1)

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address the first part of [RHINENG-7070](https://issues.redhat.com/browse/RHINENG-7070).
It implements most of the DB implementation for the GET /hosts endpoint. The 3 remaining filters are more complex, and this PR was already getting large, so I'm going to do those in a follow-up PR. 

For this PR, the DB version of the endpoint should function the same way as the XJoin version, minus the use of these params:
- `staleness`
- `registered_with`
- `filter` (i.e. System Profile filter)
- `fields` (i.e. sparse System Profile fields)

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [x] Tests: validate exceptions and failure scenarios
- [x] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [x] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by chambridge — DISMISSED** [review]
One note on `order_by` for part 2, but the rest looks good when I try the various query params locally.

File: api/host_query_db.py [hunk 1]
```diff
@@ -21,7 +33,236 @@ def get_all_hosts():
...
+    fields=None,
+    rbac_filter=None,
+) -> Tuple[List[Host], int, Tuple[str], List[str]]:
+    all_filters = _host_id_list_filter(host_id_list)
+    all_filters += _rbac_filter(rbac_filter)
+
+    return _get_host_list_using_filters(all_filters, page, per_page, param_order_by, param_order_how, fields)
+
+
+def params_to_order_by(order_by: str = None, order_how: str = None) -> Tuple:
```
[the remark this answers is not in the document]
Comment (kruai) [reply api/host_query_db.py:265]: I figured it was best to sort this now, so I implemented sorting by `group_name` and `operating_system`.

**Review by chambridge — APPROVED** [review]
lgtm

File: api/host_query_db.py [hunk 2]
```diff
@@ -21,7 +33,236 @@ def get_all_hosts():
...
+
+def _get_host_list_using_filters(
+    all_filters: List, page: int, per_page: int, param_order_by: str, param_order_how: str, fields: List[str]
+) -> Tuple[List[Host], int, Tuple[str], List[str]]:
+    host_query = _find_all_hosts().filter(*all_filters).order_by(*params_to_order_by(param_order_by, param_order_how))
+
+    query_results = host_query.paginate(page, per_page, True)
+
+    # TODO: additional_fields and system_profile_fields
+    additional_fields = tuple()
```
Comment (thearifismail) [inline api/host_query_db.py:202]: What are the additional_fields for?

File: app/models.py [hunk 3]
```diff
@@ -192,6 +197,41 @@ def _update_ansible_host(self, ansible_host):
...
+
+        if "operating_system" in self.system_profile_facts:
+            name = self.system_profile_facts["operating_system"]["name"]
+            major = self.system_profile_facts["operating_system"]["major"]
+            minor = self.system_profile_facts["operating_system"]["minor"]
+
+        return f"{name} {major:03}.{minor:03}"
+
+    @operating_system.expression
+    def operating_system(cls):
```
Comment (thearifismail) [inline app/models.py:215]: cls, what does it have/represent?

**Review by thearifismail — COMMENTED** [review]
Minor comments and questions

File: api/host_query_db.py (hunk 2 above)
Comment (kruai) [reply api/host_query_db.py:202]: This mirrors our existing XJoin-based implementation ([link](https://github.com/RedHatInsights/insights-host-inventory/blob/master/api/host_query_xjoin.py#L150)). `additional_fields` informs the serialization code which fields, in addition to the default fields, should be included in the response.

File: api/host_query_db.py [hunk 4]
```diff
@@ -12,7 +24,7 @@
 logger = get_logger(__name__)
 
 
-def get_all_hosts():
+def get_all_hosts() -> List:
```
[the remark this answers is not in the document]
Comment (kruai) [reply api/host_query_db.py:27]: The defensive dam is that we use `query.paginate` ([link to diff](https://github.com/RedHatInsights/insights-host-inventory/pull/1609/files#diff-6e8a05b37a30d3b766362c99e6ae17fbe439a1e14e1e8f1a18695115d01c7da7R199)). `paginate` limits the number of results in the query to `per_page`; its default and max value is 100.

[the remark this answers is not in the document]
Comment (thearifismail) [reply api/host_query_db.py:27]: Yes I remember now; this function was added to delete all hosts

**Review by thearifismail — APPROVED** [review]

**Pull request merged**